Explorer/Assets/DCL/Input/UnityInputSystem/Blocks/InputBlock.cs

```diff
@@ -1,37 +1,58 @@
 using Arch.Core;
-using DCL.CharacterMotion.Components;
+using DCL.Input.Component;
 using DCL.Utilities;
-using System;
+using ECS.Abstract;
+using Utility;
 
 namespace DCL.Input.UnityInputSystem.Blocks
 {
     public class InputBlock : IInputBlock
     {
-        private readonly ObjectProxy<DCLInput> dclInput;
         private readonly ObjectProxy<World> globalWorld;
-        private readonly ObjectProxy<Entity> playerEntity;
+        private SingleInstanceEntity inputMap;
 
-        public InputBlock(ObjectProxy<DCLInput> dclInput, ObjectProxy<World> globalWorld, ObjectProxy<Entity> playerEntity)
+        public InputBlock(ObjectProxy<World> globalWorld)
         {
-            this.dclInput = dclInput;
             this.globalWorld = globalWorld;
-            this.playerEntity = playerEntity;
         }
 
-        public void BlockMovement()
+        public void Initialize()
         {
-            globalWorld.StrictObject.AddOrGet(playerEntity.StrictObject, new MovementBlockerComponent());
-            dclInput.StrictObject.Shortcuts.Disable();
-            dclInput.StrictObject.Camera.Disable();
-            dclInput.StrictObject.Player.Disable();
+            inputMap = globalWorld.StrictObject.CacheInputMap();
         }
 
-        public void UnblockMovement()
+        public void BlockInputs(InputMapComponent.Kind kinds, bool singleValue = false)
         {
-            globalWorld.StrictObject.Remove<MovementBlockerComponent>(playerEntity.StrictObject);
-            dclInput.StrictObject.Shortcuts.Enable();
-            dclInput.StrictObject.Camera.Enable();
-            dclInput.StrictObject.Player.Enable();
+            ref var inputMapComponent = ref inputMap.GetInputMapComponent(globalWorld.StrictObject);
+
+            if (singleValue)
+            {
+                inputMapComponent.BlockInput(kinds);
+                return;
+            }
+
+            for (var i = 0; i < InputMapComponent.VALUES.Count; i++)
+            {
+                InputMapComponent.Kind kind = InputMapComponent.VALUES[i];
+                if (EnumUtils.HasFlag(kinds, kind)) { inputMapComponent.BlockInput(kind); }
+            }
+        }
+
+        public void UnblockInputs(InputMapComponent.Kind kinds, bool singleValue = false)
+        {
+            ref var inputMapComponent = ref inputMap.GetInputMapComponent(globalWorld.StrictObject);
```

Comment:
maybe if you make it via paramas, then you can just iterate over it and take values directly from dictionary, instead of iterating dictionary. It will probably allow you to remove `singleValue` block of code. Same applies to `BlockInput` method

Reply:
yeah I was thinking the same at some point :) seems like a good approach!